test/sharness/t0080-repo.sh

```diff
@@ -152,6 +152,31 @@ test_expect_success "'ipfs pin ls --type=all --quiet' is correct" '
 	test_sort_cmp allpins_uniq_hashes actual_allpins
 '
 
+test_expect_success "'ipfs refs --unique' is correct" '
+	mkdir -p uniques &&
+	cd uniques &&
+	echo "content1" > file1 &&
+	echo "content1" > file2 &&
+	ROOT=`ipfs add -r -q . | tail -n 1` &&
+	ipfs refs --unique $ROOT > expected &&
+	ipfs add -q file1 > unique_hash &&
+	test_cmp expected unique_hash
+'
+
+test_expect_success "'ipfs refs --unique --recursive' is correct" '
+	rm * &&
+	mkdir -p sub1/sub2 &&
+	echo "content1" > file1 &&
+	echo "content1" > sub1/file1 &&
+	echo "content1" > sub1/sub2/file1 &&
+	echo "content2" > sub1/sub2/file2 &&
+	ROOT=`ipfs add -r -q . | tail -n 1` &&
+	ipfs refs --unique --recursive $ROOT > refs_output &&
+	wc -l refs_output > line_count &&
+	echo "4 refs_output" > expected &&
+	test_cmp line_count expected
+'
+
```

Comment:
Hey I could absolutely add more tests. This was just the bare minimum to ensure that the ticket was actually closeable :)

That stated, I don't think this line will work `ipfs refs -r -u $root >unique_actual_sorted &&` because `refs -r -u` does not actually sort. It drills in to the structure and outputs as it goes. This was actually the bug, the code was escaping early on a duplicate because it assumed the duplicate was  'folder' instead of a 'file'.